xformers/components/attention/csrc/cuda/mem_eff_attention/kernel_backward.h

```diff
@@ -1070,6 +1241,39 @@ struct AttentionBackwardKernel {
       auto output_tile_coords = cutlass::MatrixCoord{
           warp_idx_mn_0 % Mma::Base::WarpCount::kM,
           warp_idx_mn_0 / Mma::Base::WarpCount::kM};
+
+      // apply bias if applicable
+      if (p.bias_ptr != nullptr) {
+        // load bias tile Bij into shared memory
+        typename MatmulQK::BiasLoader::GmemTileIterator bias_iter(
+            {cutlass::layout::RowMajor(p.bias_strideM)},
+            p.bias_ptr + query_start * p.bias_strideM + key_start,
+            {num_queries_in_block, num_keys_in_block},
+            thread_id);
+        cutlass::TensorRef<scalar_t, cutlass::layout::RowMajor> bias_tensor_ref(
+            shared_storage.bias().data(),
+            cutlass::layout::RowMajor(MatmulQK::ThreadblockShape::kM));
+        typename MatmulQK::BiasLoader::SmemTileIterator smem_tile_iter(
+            bias_tensor_ref, thread_id);
+        MatmulQK::BiasLoader::load(bias_iter, smem_tile_iter);
+
+        // Pij += Bij, where Pij is in register fragment and Bij is in shared memory
+        auto lane_offset = MatmulQK::ScalingCoefsUpdater::get_lane_offset(
+            lane_id, warp_id, output_tile_coords);
+        MatmulQK::ScalingCoefsUpdater::iterateCols(
```

Comment:
yeah the hope was to avoid some bank conflicts, and was a bit faster in my testing. this is one of the older changes though so let me go back and make sure its still needed. will comment if it stays.

also just realized i forgot to implement iterateCols for volta, so will add that if needed too.

Reply:
"It's faster" is a good enough explanation for me ^^

> also just realized i forgot to implement iterateCols for volta, so will add that if needed too.

We will need to have support for all architectures we support yes

Reply:
tested again and it didn't make a difference, not sure why it made a difference before. got rid of it